pkg/aws/snow.go

```diff
@@ -0,0 +1,56 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+const (
+	snowEC2Port = 8243
+)
+
+func snowEndpoint(deviceIP string) *ServiceEndpoint {
+	return &ServiceEndpoint{
+		ServiceID:     "EC2",
+		SigningRegion: "snow",
+		URL:           fmt.Sprintf("https://%s:%d", deviceIP, snowEC2Port),
+	}
+}
+
+func WithCustomCABundleFile(certsFile string) config.LoadOptionsFunc {
+	caPEM, err := ioutil.ReadFile(certsFile)
+	if err != nil {
+		return func(*config.LoadOptions) error {
+			return fmt.Errorf("reading aws certificates file: %v", err)
+		}
+	}
+
+	return config.WithCustomCABundle(bytes.NewReader(caPEM))
```

Comment:
This is OK but there is a nuance in behavior from what a maintainer likely intuitively expects.

Intuitively I feel a maintainer, given the `WithCustomCABundleFile` doesn't return an error, would assume the file is read _when the option is applied_. That gives cause for me to, hypothetically, write a file _after_ calling `WithCustomCABundleFile` but _before_ the option is applied. While that may seem silly, its quite legitimate when reflecting on the package API.

Consequently, is it possible for us to defer reading the file until we actually apply the option so we don't have any potential gotchas? See below.

```suggestion
	return func(opts *config.LoadOptions) error {
		caPEM, err := ioutil.ReadFile(certsFile)
		if err != nil {
			return fmt.Errorf("reading aws certificates file: %v", err)
		}
		
		customBundleOpt := config.WithCustomCABundle(bytes.NewReader(caPEM))
		
		if err := customBundleOpt(opts); err != nil {
			return err
		}
		
		return 
	}
```